azure/iac/sql_servers_auditing.rego

```diff
@@ -0,0 +1,55 @@
+package rule
+
+# https://docs.microsoft.com/en-us/azure/templates/microsoft.sql/2021-02-01-preview/servers/auditingsettings
+
+#
+# PR-AZR-0134-ARM
+#
+
+default sql_server_log_audit = null
+
+azure_attribute_absence["sql_server_log_audit"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.sql/servers/auditingsettings"
+    not resource.properties.state
+}
+
+azure_issue["sql_server_log_audit"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.sql/servers/auditingsettings"
+    lower(resource.properties.state) != "enabled"
```

Comment:
**We might need to consider the description of the state** at https://docs.microsoft.com/en-us/azure/templates/microsoft.sql/2021-02-01-preview/servers/auditingsettings?tabs=bicep
Is there any way we can check  `storageEndpoint `or `isAzureMonitorTargetEnabled `is avaiable? i am not sure that much check is necessary, it will fail eventually during deploy if dependent resource not available with state enabled.
@farchide thought?
<html>
<body>
<!--StartFragment-->

state | Specifies the state of the audit. If state is Enabled, storageEndpoint or isAzureMonitorTargetEnabled are required.
-- | --


<!--EndFragment-->
</body>
</html>